### ISSUE TITLE
Look into performance improvements

### ISSUE THREAD

**Issue by satya164** [callstack-io/haul#95]

**Comment by axemclion** [comment]
I did some performance comparisons - http://blog.nparashuram.com/2017/04/react-native-packager-vs-haul.html

**Comment by grabbou** [comment]
Thanks, I just had a quick read to see how it performs. Not that bad as far as first builds are concerned. We've already made few improvements that are not yet released and are likely to make the initial difference smaller. 

Is the repository you linked on Twitter earlier the source code you used for tests? I would be interested in having a framework to continuously test how it performs.

**Comment by satya164** [comment]
Maybe we need some tests on CI which report back the change in performance.

**Comment by axemclion** [comment]
@grabbou Yes it is - The repo is at https://github.com/nparashuram/Haul-vs-ReactNative. 

[![asciicast](https://asciinema.org/a/3hxgz9629wufaxle14q9j8zi6.png)](https://asciinema.org/a/3hxgz9629wufaxle14q9j8zi6)
